chore: update pg_upgrade docs around incompatible extension versions; remove plv8 object exception (#20526)
supabase/supabase: apps/docs/pages/guides/platform/migrating-and-upgrading-projects.mdx

pg_upgrade does not support upgrading of databases containing reg\* data types referencing system OIDs.
If you have created any objects that depend on the following extensions, you will need to recreate them after the upgrade.

#### Extensions

> pg_upgrade does not currently support upgrading of databases using extensions older than the following versions:

Author: Alternative wording, though it might prove confusing for users skimming the docs, leading them to believe that projects using those specific versions can't be upgraded, rather than the reverse

```suggestion
pg_upgrade does not currently support upgrading of databases using extensions prior to following versions:
```
